File: backend/app/domain/repositories/growth_repository.py
from abc import ABC, abstractmethod
from uuid import UUID

from app.domain.entities.growth_record import GrowthRecord


class GrowthRepository(ABC):
    @abstractmethod
    async def get(self, id: UUID) -> GrowthRecord | None:
        ...

    @abstractmethod
    async def get_by_baby_id(self, baby_id: UUID) -> list[GrowthRecord]:
        ...

    @abstractmethod
    async def save(self, record: GrowthRecord) -> GrowthRecord:
        ...

    @abstractmethod
    async def delete(self, id: UUID) -> None:
        ...

File: zzippu_ios/zzippu/Domain/Repositories/GrowthRepository.swift
// Domain/Repositories/GrowthRepository.swift
// Foundation only — SwiftUI/SwiftData import 금지

import Foundation

protocol GrowthRepository {
    func create(_ record: GrowthRecord) async throws -> GrowthRecord
    func update(_ record: GrowthRecord) async throws -> GrowthRecord
    func delete(id: UUID, babyId: UUID) async throws
    func series(babyId: UUID) async throws -> [GrowthRecord]
}
